dashboard/api.py

```diff
@@ -583,7 +584,20 @@ def has_to_pay_for_exam(mmtrack, course):
     Returns:
         bool: if the user has to pay for another exam attempt
     """
-    attempt_limit = mmtrack.get_payments_count_for_course(course) * ATTEMPTS_PER_PAID_RUN
+    now = now_in_utc()
+    attempt_limit = 0
+    num_attempts = ATTEMPTS_PER_PAID_RUN
+    if not mmtrack.program.exam_attempts_first_date:
+        num_attempts = ATTEMPTS_PER_PAID_RUN_OLD
+    elif mmtrack.program.exam_attempts_first_date > now:
+        # still before the date
+        num_attempts = ATTEMPTS_PER_PAID_RUN_OLD
+    elif mmtrack.program.exam_attempts_second_date > now:
+        # in between the dates: check when the user paid for each course run
+        attempt_limit = mmtrack.get_custom_number_of_attempts_for_course(course)
+
+    if not attempt_limit:
```

Comment:
What if `attempt_limit` is `0`?

Reply:
If it is 0 that means that the number of total exams attempts can be computed by multiplying the number of payments by the number of attempts per run. 
In the case where we are in between the dates, Peter asked to give 2 attempts to users who paid before the first date, and 1 attempt after the first date, so that is computed in `get_custom_number_of_attempts_for_course`.

Reply:
Can we change `attempt_limit = None` and have this be `if attempt_limit is None`? In general we should use `None` rather than `0` as a sentinel value